### ISSUE TITLE
Watchboy error after change of file

### ISSUE THREAD

**Issue by jabdr** [catdad/electronmon#60]
I've just added electronmon to a new react+webpack+typescript+electron app. Unfortunatly if I run it with electronmon the process crashes with the following output:

```sh
!> yarn mon
yarn run v1.22.5
$ electronmon ./dist/main.bundle.js

[electronmon] waiting for a change to restart it
(node:6564) ExtensionLoadWarning: Warnings loading extension at C:\Users\jdrum\AppData\Roaming\Electron\extensions\fmkadmapgofadopljbjfkapdkoienihi: Unrecognized manifest key 'browser_action'. Unrecognized manifest key 'minimum_chrome_version'. Unrecognized manifest key 'update_url'. Cannot load extension with file or directory name _metadata. Filenames starting with "_" are reserved for use by the system.
Added Extension:  React Developer Tools
(node:6564) ExtensionLoadWarning: Warnings loading extension at C:\Users\jdrum\AppData\Roaming\Electron\extensions\lmhkpmbekcpmknklioeibfkpmmfibljd: Unrecognized manifest key 'commands'. Unrecognized manifest key 'homepage_url'. Unrecognized manifest key 'page_action'. Unrecognized manifest key 'short_name'. Unrecognized manifest key 'update_url'. Permission 'notifications' is unknown or URL pattern is malformed. Permission 'contextMenus' is unknown or URL pattern is malformed. Permission 'tabs' is unknown or URL pattern is malformed. Cannot load extension with file or directory name _metadata. Filenames starting with "_" are reserved for use by the system.
Added Extension:  Redux DevTools
[electronmon] renderer file change: dist\renderer.bundle.js
[electronmon] renderer file change: dist\renderer.bundle.js.map
C:\Users\jdrum\Documents\Projects\birder\node_modules\watchboy\index.js:182
      pending[abspath].timer = -1;
                             ^

TypeError: Cannot set property 'timer' of undefined
    at Timeout._onTimeout (C:\Users\jdrum\Documents\Projects\birder\node_modules\watchboy\index.js:182:30)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

OS: Windows 10
Node: > v15.11.0

```json
  "dependencies": {
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-router": "^5.2.0",
    "react-router-dom": "^5.2.0"
  },
  "devDependencies": {
    "@types/mocha": "^8.2.0",
    "@types/react": "^17.0.0",
    "@types/react-dom": "^17.0.0",
    "@types/react-router": "^5.1.8",
    "@types/react-router-dom": "^5.1.6",
    "@typescript-eslint/eslint-plugin": "^4.11.0",
    "@typescript-eslint/parser": "^4.11.0",
    "copy-webpack-plugin": "^7.0.0",
    "cross-env": "^7.0.3",
    "css-loader": "^5.0.1",
    "electron": "^11.1.1",
    "electron-builder": "^22.9.1",
    "electron-devtools-installer": "^3.1.1",
    "electronmon": "^2.0.0",
    "eslint": "^7.16.0",
    "eslint-config-airbnb": "^18.2.1",
    "eslint-import-resolver-webpack": "^0.13.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-react": "^7.21.5",
    "eslint-plugin-react-hooks": "^4.2.0",
    "file-loader": "^6.2.0",
    "html-webpack-plugin": "^4.5.0",
    "lodash": "^4.17.20",
    "mocha": "^8.2.1",
    "npm-run-all": "^4.1.5",
    "react-devtools": "^4.10.1",
    "rimraf": "^3.0.2",
    "serve": "^11.3.2",
    "source-map-loader": "^2.0.0",
    "spectron": "^13.0.0",
    "style-loader": "^2.0.0",
    "ts-loader": "^8.0.12",
    "ts-node": "^9.1.1",
    "tsconfig-paths": "^3.9.0",
    "typescript": "^4.1.3",
    "webpack": "^5.11.0",
    "webpack-cli": "^4.3.0"
  },
```

First I thought that maybe the changed ts files cause trouble because of webpack, so I've excluded src folder with no luck:
```json
  "electronmon": {
    "patterns": ["!src/**", "dist/**"]
  }
```

Could you please help me to fix this issue?

**Comment by catdad** [comment]
Sorry for the delay, I didn't see this until now.

This was reported in https://github.com/catdad/watchboy/issues/20, shouldn't be too hard to fix. I'll take a look.

**Comment by catdad** [comment]
`watchboy@0.4.3` should fix this issue. You should automatically get that version if you just reinstall `electronmon`. I'll make this the new required minimum version later today as well.

**Comment by catdad** [comment]
This should be fixed in `electronmon@2.0.1`
